src/main/java/cubicchunks/server/chunkio/IONbtWriter.java

```diff
@@ -63,25 +68,31 @@
     }
 
     static NBTTagCompound write(Column column) {
-        NBTTagCompound nbt = new NBTTagCompound();
-        writeBaseColumn(column, nbt);
-        writeBiomes(column, nbt);
-        writeOpacityIndex(column, nbt);
-        return nbt;
+        NBTTagCompound columnNbt = new NBTTagCompound();
+        NBTTagCompound level = new NBTTagCompound();
+        columnNbt.setTag("Level", level);
+        columnNbt.setInteger("DataVersion", MINECRAFT_CHUNK_DATA_VERSION);
+        FMLCommonHandler.instance().getDataFixer().writeVersionData(columnNbt);
+        writeBaseColumn(column, level);
+        writeBiomes(column, level);
+        writeOpacityIndex(column, level);
+        MinecraftForge.EVENT_BUS.post(new ChunkDataEvent.Save(column, columnNbt));
+        return columnNbt;
     }
 
     static NBTTagCompound write(final Cube cube) {
         NBTTagCompound cubeNbt = new NBTTagCompound();
-
-        writeBaseCube(cube, cubeNbt);
-
-        writeBlocks(cube, cubeNbt);
-        writeEntities(cube, cubeNbt);
-        writeTileEntities(cube, cubeNbt);
-        writeScheduledTicks(cube, cubeNbt);
-
-        writeLightingInfo(cube, cubeNbt);
-
+        //Added to preserve compatibility with vanilla NBT chunk format.
+        NBTTagCompound level = new NBTTagCompound();
+        cubeNbt.setTag("Level", level);
+        cubeNbt.setInteger("DataVersion", MINECRAFT_CHUNK_DATA_VERSION);
+        FMLCommonHandler.instance().getDataFixer().writeVersionData(cubeNbt);
+        writeBaseCube(cube, level);
+        writeBlocks(cube, level);
```

Comment:
To make sure things are compatible, cube data probably needs to be in an array, to pretend that a Cube is actually a chunk with only one section.